otter/controller.py

```diff
@@ -348,3 +371,88 @@ def calculate_delta(log, state, config, policy):
     """
     current = len(state.active) + len(state.pending)
     return apply_delta(log, current, state, config, policy)
+
+
+@do
+def _is_server_in_group(group, server_id):
+    """
+    Given a group and server ID, determines if the server is a member of
+    the group.  If it isn't, it raises a :class:`ServerNotFoundError`.
+    """
+    try:
+        server_info = yield retry_effect(get_server_details(server_id),
+                                         retry_times(3),
+                                         exponential_backoff_interval(2))
+    except NoSuchServerError:
+        raise ServerNotFoundError(group.tenant_id, group.uuid, server_id)
+
+    group_id = group_id_from_metadata(
+        get_in(('server', 'metadata'), server_info, {}))
+    if group_id != group.uuid:
+        raise ServerNotFoundError(group.tenant_id, group.uuid, server_id)
+
+
+@do
+def _can_maybe_scale_down(group, server_id):
```

Comment:
Nitpick: "can" and "maybe" are probably repetetive. You could just have `_can_scale_down`?